cborg-json/src/Codec/CBOR/JSON.hs

```diff
@@ -44,24 +46,33 @@ decodeValue :: Bool -> Decoder s Value
 decodeValue lenient = do
     tkty <- peekTokenType
     case tkty of
-      TypeUInt    -> decodeNumberIntegral
-      TypeUInt64  -> decodeNumberIntegral
-      TypeNInt    -> decodeNumberIntegral
-      TypeNInt64  -> decodeNumberIntegral
-      TypeInteger -> decodeNumberIntegral
-      TypeFloat16 -> decodeNumberFloat16
-      TypeFloat32 -> decodeNumberFloating
-      TypeFloat64 -> decodeNumberFloating
-      TypeBool    -> Bool   <$> decodeBool
-      TypeNull    -> Null   <$  decodeNull
-      TypeString  -> String <$> decodeString
-
+      TypeUInt         -> decodeNumberIntegral
+      TypeUInt64       -> decodeNumberIntegral
+      TypeNInt         -> decodeNumberIntegral
+      TypeNInt64       -> decodeNumberIntegral
+      TypeInteger      -> decodeNumberIntegral
+      TypeFloat16      -> decodeNumberFloat16
+      TypeFloat32      -> decodeNumberFloating
+      TypeFloat64      -> decodeNumberFloating
+      TypeBool         -> Bool   <$> decodeBool
+      TypeBreak        -> Bool   <$> decodeBreakOr
+      TypeBytes        -> decodeByteString
+      TypeBytesIndef   -> decodeByteStringIndef []
+      TypeNull         -> Null   <$  decodeNull
+      TypeSimple       -> decodeSimple'
+      TypeString       -> String <$> decodeString
+      TypeStringIndef  -> decodeStringIndef' []
+      TypeTag          -> decodeTag'
+      TypeTag64        -> decodeTag64'
       TypeListLen      -> decodeListLen >>= decodeListN lenient
+      TypeListLen64    -> decodeListLen >>= decodeListN lenient
       TypeListLenIndef -> decodeListLenIndef >> decodeListIndef lenient []
       TypeMapLen       -> decodeMapLen >>= flip (decodeMapN lenient) HM.empty
+      TypeMapLen64     -> decodeMapLen >>= flip (decodeMapN lenient) HM.empty
+      TypeMapLenIndef  -> decodeMapLenIndef >> decodeMapIndefLen lenient HM.empty
+      TypeInvalid      -> fail $ "Unexpected CBOR token type for a JSON value: "
+                               <> show tkty
```

Comment:
The thing to note is that there's actually several CBOR constructs that cannot be converted into JSON. So the original catch-all was catching all the ones that could not be converted, plus a bunch more that were added later that could be converted.